istioctl/cmd/analyze.go

```diff
@@ -161,3 +168,24 @@ func serviceDiscovery() (bool, error) {
 		return false, fmt.Errorf("invalid argument value for discovery")
 	}
 }
+
+func errorIfMessagesExceedThreshold(messages []diag.Message) error {
+	foundIssues := false
+	for _, m := range messages {
+		if m.Type.Level().IsWorseThanOrEqualTo(messageLevelThreshold) {
+			foundIssues = true
+		}
+	}
+
+	if foundIssues {
+		return AnalyzerFoundIssuesError{}
+	}
+
+	return nil
+}
+
+func outputMessages(f io.Writer, messages []diag.Message) {
+	for _, m := range messages {
+		fmt.Fprintf(f, "%v\n", m.String())
+	}
+}
```

Comment:
As a quick & dirty solution for https://github.com/istio/istio/issues/17996, can we add something like this to the beginning of the fn?:

```
if len(messages) == 0 {
    fmt.Fprintf(f, "No validation issues found")
    return
}
```

Reply:
Done and done.